packages/sharing-smart-contract/contracts/ProtectedDataSharing.sol

```diff
@@ -202,13 +211,19 @@ contract ProtectedDataSharing is ERC721Burnable, ERC721Receiver, ManageOrders, A
     //protectedData's owner should approve this SC
     function addProtectedDataToCollection(
         uint256 _collectionId,
-        address _protectedData
+        address _protectedData,
+        address _appAddress
     ) public onlyCollectionOwner(_collectionId) {
+        require(_appAddress != address(0), "App address invalid");
```

Comment:
this check seems useless, any valid address that is not the address of an app owned by the contract is problematic.

Reply:
Yeah but i think it's the vital minimum to check. May we should check that the ownership of the app has been transferred to this SC

Reply:
Commit: 8cd1186ee574f5d2f8ca006f7d70d68f4160f89d